python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: The preprocessor used to transform data batches prior
+            to prediction.
+
+    """
+
+    def __init__(self, model: torch.nn.Module, preprocessor: Preprocessor):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition: If checkpoints
+                contains the model state dict, and not the model itself,
+                then the state dict will be loaded to the model_definition.
+
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        column_names: Optional[List[str]] = None,
+        feature_columns: Optional[Union[List[str], List[List[str]]]] = None,
+        dtype: Optional[Union[torch.dtype, List[torch.dtype]]] = None,
```

Comment:
why can't these go into preprocessor? or the model itself (as first layer or in its call function)

Reply:
here we are overloading the predict interface, should we go back and add a **kwargs to the parent interface? (for example, in Serve's model wrapper, we expect Predictor.predict only take a single argument)

Reply:
ah yeah **kwargs was already to Predictor.predict in a previous PR (https://github.com/ray-project/ray/commit/79a3b56015586a2a825ca52cc85184a0b47979a0)

Reply:
@simon-mo can you raise the first point in the broader group for more visibility/discussion?

Reply:
ahhh got it. this resolved my second question

Reply:
(Resolved offline)